src/avlizer.erl

```diff
@@ -25,17 +25,72 @@
         ]).
 
 start(_StartType, _StartArgs) ->
-  case osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_URL", false) of
-    false -> ok;
-    URL ->
-      Vars0 = application:get_env(?APPLICATION, avlizer_confluent, #{}),
-      Vars = Vars0#{schema_registry_url => URL},
-      application:set_env(?APPLICATION, avlizer_confluent, Vars)
-  end,
+  set_envs(),
   avlizer_sup:start_link().
 
 stop(_State) -> ok.
 
+set_envs() ->
+  Vars0 = application:get_env(?APPLICATION, avlizer_confluent, #{}),
+  Vars1 = maybe_put_schema_registry_url(Vars0),
+  Vars2 = maybe_put_schema_registry_auth(Vars1),
+  application:set_env(?APPLICATION, avlizer_confluent, Vars2).
+
+maybe_put_schema_registry_url(Map) ->
+  SchemaRegistryURL = osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_URL", false),
+  case SchemaRegistryURL of
+    false -> Map;
+    URL -> maps:put(schema_registry_url, URL, Map)
+  end.
+
+maybe_put_schema_registry_auth(Map) ->
+  case osenv_all([
+   "AVLIZER_CONFLUENT_SCHEMAREGISTRY_AUTH_MECHANISM",
+   "AVLIZER_CONFLUENT_SCHEMAREGISTRY_AUTH_USERNAME",
+   "AVLIZER_CONFLUENT_SCHEMAREGISTRY_AUTH_PASSWORD"
+  ], false) of
+    [Mechanism, Username, Password] -> 
+      maps:put(schema_registry_auth, {list_to_atom(Mechanism), Username, Password}, Map);
+    false -> 
+      maybe_read_auth_file(Map)
+  end.
+
+maybe_read_auth_file(#{schema_registry_auth := {_Mechanism, File}} = Vars) ->
+  case file:read_file(File) of
+    {ok, Binary} -> maybe_read_auth_file(Vars, Binary);
+    Error -> Error
+  end;
+maybe_read_auth_file(Vars) -> Vars.
+
+maybe_read_auth_file(#{schema_registry_auth := {Mechanism, File}} = Vars, Binary) ->
+  Lines = binary:split(Binary, <<"\n">>, [trim_all, global]),
+  case Lines of
+    [Username, Password] -> 
+      Auth = {Mechanism, binary_to_list(Username), binary_to_list(Password)},
+      maps:put(schema_registry_auth, Auth, Vars);
+    _Malformed -> 
+      error_logger:error_msg("Malformed authorization file ~s", [File]),
```

Comment:
This message can be changed as:

`"~p: Malformed authorization file: ~s", [?MODULE, File]` 

to help troubleshooting